scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -1236,47 +1417,62 @@ def test_getitem():
         [0, 0, -1],
         [0, 1, 0]
     ])
+    mat = xp.asarray(mat)
     r = Rotation.from_matrix(mat)
 
-    assert_allclose(r[0].as_matrix(), mat[0], atol=1e-15)
-    assert_allclose(r[1].as_matrix(), mat[1], atol=1e-15)
-    assert_allclose(r[:-1].as_matrix(), np.expand_dims(mat[0], axis=0), atol=1e-15)
+    xp_assert_close(r[0].as_matrix(), mat[0], atol=1e-15)
+    xp_assert_close(r[1].as_matrix(), mat[1, ...], atol=1e-15)
+    xp_assert_close(r[:-1].as_matrix(), xp.expand_dims(mat[0, ...], axis=0), atol=1e-15)
 
 
-def test_getitem_single():
+def test_getitem_single(xp):
     with pytest.raises(TypeError, match='not subscriptable'):
-        Rotation.identity()[0]
+        Rotation.from_quat(xp.asarray([0, 0, 0, 1]))[0]
 
 
-def test_setitem_single():
-    r = Rotation.identity()
+def test_getitem_array_like(xp):
+    mat = np.array([[[0.0, -1, 0],
+                     [1, 0, 0],
+                     [0, 0, 1]],
+
+                    [[1, 0, 0],
+                     [0, 0, -1],
+                     [0, 1, 0]]])
+    mat = xp.asarray(mat)
+    r = Rotation.from_matrix(mat)
+    xp_assert_close(r[[0]].as_matrix(), mat[[0]], atol=1e-15)
+    xp_assert_close(r[[0, 1]].as_matrix(), mat[[0, 1]], atol=1e-15)
```

Comment:
I'd rather leave this behaviour untested for xp != np. Intuitively, this should be aligned to the Array API `__getitem__` where array-likes are not supported (read: their behaviour is undefined).
```suggestion
def test_getitem_array_like():
    mat = np.array([[[0.0, -1, 0],
                     [1, 0, 0],
                     [0, 0, 1]],
                    [[1, 0, 0],
                     [0, 0, -1],
                     [0, 1, 0]]])
    r = Rotation.from_matrix(mat)
    xp_assert_close(r[[0]].as_matrix(), mat[[0]], atol=1e-15)
    xp_assert_close(r[[0, 1]].as_matrix(), mat[[0, 1]], atol=1e-15)
```